hack/ci/e2e-k8s.sh

```diff
@@ -13,303 +13,184 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# hack script for running a kind e2e
-# must be run with a kubernetes checkout in $PWD (IE from the checkout)
-# Usage: SKIP="ginkgo skip regex" FOCUS="ginkgo focus regex" kind-e2e.sh
-
 set -o errexit -o nounset -o xtrace
 
-# Settings:
-# SKIP: ginkgo skip regex
-# FOCUS: ginkgo focus regex
-# LABEL_FILTER: ginkgo label query for selecting tests (see "Spec Labels" in https://onsi.github.io/ginkgo/#filtering-specs)
-#
-# The default is to focus on conformance tests. Serial tests get skipped when
-# parallel testing is enabled. Using LABEL_FILTER instead of combining SKIP and
-# FOCUS is recommended (more expressive, easier to read than regexp).
-#
-# GA_ONLY: true  - limit to GA APIs/features as much as possible
-#          false - (default) APIs and features left at defaults
-# FEATURE_GATES:
-#          JSON or YAML encoding of a string/bool map: {"FeatureGateA": true, "FeatureGateB": false}
-#          Enables or disables feature gates in the entire cluster.
-#          Cannot be used when GA_ONLY=true.
-# RUNTIME_CONFIG:
-#          JSON or YAML encoding of a string/string (!) map: {"apia.example.com/v1alpha1": "true", "apib.example.com/v1beta1": "false"}
-#          Enables API groups in the apiserver via --runtime-config.
-#          Cannot be used when GA_ONLY=true.
+SKIP="${SKIP:-}"
+FOCUS="${FOCUS:-}"
+LABEL_FILTER="${LABEL_FILTER:-}"
+GA_ONLY="${GA_ONLY:-false}"
+FEATURE_GATES="${FEATURE_GATES:-{\}}"
+RUNTIME_CONFIG="${RUNTIME_CONFIG:-{\}}"
+KIND_CLUSTER_LOG_LEVEL="${KIND_CLUSTER_LOG_LEVEL:-4}"
+CLUSTER_LOG_FORMAT="${CLUSTER_LOG_FORMAT:-}"
+KUBELET_LOG_FORMAT="${KUBELET_LOG_FORMAT:-${CLUSTER_LOG_FORMAT}}"
+IP_FAMILY="${IP_FAMILY:-ipv4}"
+KUBE_PROXY_MODE="${KUBE_PROXY_MODE:-iptables}"
+PARALLEL="${PARALLEL:-false}"
+
+log() {
+    echo "[$(date +'%Y-%m-%d %H:%M:%S')] $*"
+}
 
-# cleanup logic for cleanup on exit
-CLEANED_UP=false
 cleanup() {
-  if [ "$CLEANED_UP" = "true" ]; then
-    return
-  fi
-  # KIND_CREATE_ATTEMPTED is true once we: kind create
-  if [ "${KIND_CREATE_ATTEMPTED:-}" = true ]; then
-    kind "export" logs "${ARTIFACTS}" || true
-    kind delete cluster || true
-  fi
-  rm -f _output/bin/e2e.test || true
-  # remove our tempdir, this needs to be last, or it will prevent kind delete
-  if [ -n "${TMP_DIR:-}" ]; then
-    rm -rf "${TMP_DIR:?}"
-  fi
-  CLEANED_UP=true
+    if [ "${CLEANED_UP:-false}" = "true" ]; then
```

Comment:
There are a lot of unnecessary whitespace changes included that make this difficult to review.